### PR TITLE
Finish application translation

### LINKED ISSUE #8: Finish application internationalization
Right now the application defines just four translations keys. See:
https://github.com/symfony/symfony-demo/blob/master/app/Resources/translations/messages.en.xliff

We did this to have just a simple example of how to translate application interfaces using Symfony. I propose the following:

  * Extract all the strings from the public interface and add them to the `messages.en.xliff` file
  * Move all the backend related strings to the `admin.en.xliff` file to show an example of using translation catalogues
  * Once these tasks are finished, ask the community to contribute translations.

What do you think?

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #8 thanks to the work made by @b-durand in #78.

### CODE REVIEW

**Comment by javiereguiluz** [comment]
The reorganization of the English translation is now complete. Please review it before I make the needed changes in the rest of translations. Thanks.

**Comment by OskarStark** [comment]
@javiereguiluz if this is merged i will do the german translation if you like and if useful

**Comment by javiereguiluz** [comment]
@OskarStark great! That would be awesome.

**Comment by OskarStark** [comment]
ok :smile:

**Comment by javiereguiluz** [comment]
I have a question about the names of the translation strings related to the entity validation.

I've defined them as follows, but I don't like it:

```xml
<trans-unit id="entity.comment_blank">
    <source>entity.comment_blank</source>
    <target>Please don't leave your comment blank!</target>
</trans-unit>
<trans-unit id="entity.comment_short">
    <source>entity.comment_short</source>
    <target>Comment is too short ({{ limit }} characters minimum)</target>
</trans-unit>
<trans-unit id="entity.comment_long">
    <source>entity.comment_long</source>
    <target>Comment is too long ({{ limit }} characters maximum)</target>
</trans-unit>
<trans-unit id="entity.comment_invalid">
    <source>entity.comment_invalid</source>
    <target>The content of this comment is considered spam.</target>
</trans-unit>
<trans-unit id="entity.summary_blank">
    <source>entity.summary_blank</source>
    <target>Give your post a summary!</target>
</trans-unit>
<trans-unit id="entity.post_short">
    <source>entity.post_short</source>
    <target>Post content is too short ({{ limit }} characters minimum)</target>
</trans-unit>
```

What do you think of replacing it by the following?

```xml
<trans-unit id="comment.blank">
    <source>comment.blank</source>
    <target>Please don't leave your comment blank!</target>
</trans-unit>
<trans-unit id="comment.too_short">
    <source>comment.too_short</source>
    <target>Comment is too short ({{ limit }} characters minimum)</target>
</trans-unit>
<trans-unit id="comment.too_long">
    <source>comment.too_long</source>
    <target>Comment is too long ({{ limit }} characters maximum)</target>
</trans-unit>
<trans-unit id="comment.is_spam">
    <source>comment.is_spam</source>
    <target>The content of this comment is considered spam.</target>
</trans-unit>
<trans-unit id="comment.blank_summary">
    <source>comment.blank_summary</source>
    <target>Give your post a summary!</target>
</trans-unit>
<trans-unit id="post.too_short">
    <source>post.too_short</source>
    <target>Post content is too short ({{ limit }} characters minimum)</target>
</trans-unit>
```

**Comment by OskarStark** [comment]
:+1:  for shorter version!

**Comment by b-durand** [comment]
@javiereguiluz the domain is validators and not message for constraints.

**Comment by javiereguiluz** [comment]
@b-durand I can't believe I made this mistake :'(  This morning I saw that translation wasn't working for validation ... but I didn't realize this issue. Thanks for telling me! It's fixed now.

**Comment by b-durand** [comment]
We should also localize the date format for "commented on". I regret to have choice with only 4 formats instead of format pattern for skeleton :disappointed:

**Comment by javiereguiluz** [comment]
@b-durand I'm not sure I understand your last comment about date internationalization.

**Comment by OskarStark** [comment]
i think @b-durand would say, that in englisch we use ```01-01-2015``` and in german ```01.01.2015````

**Comment by javiereguiluz** [comment]
OK, I understand now. Thanks. Let's do that change in a separate PR because it needs to install the Twig extensions and all that.

**Comment by javiereguiluz** [comment]
@b-durand this feature is finally merged. Thanks a lot for working on this!

**Comment by javiereguiluz** [comment]
@OskarStark this feature is finally merged, so you can update the German translation if you have some free time. Thanks!

File: src/AppBundle/Form/PostType.php [hunk 1]
```diff
@@ -32,14 +32,16 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         // for the full reference of options defined by each form field type
         // see http://symfony.com/doc/current/reference/forms/types.html
         $builder
-            ->add('title')
-            ->add('summary', 'textarea')
+            ->add('title', null, ['label' => 'label.title'])
```
Comment (Tobion) [inline src/AppBundle/Form/PostType.php]: this is not compatible with php 5.3 (as stated in composer.json). so I'd suggest increasing the php requirement for the demo. we do not need to support 5.3.

Comment (javiereguiluz) [inline src/AppBundle/Form/PostType.php]: Good catch! I'm going to check of non-PHP 5.3 features and I'm going to setup a `.travis.yml` file so we can check on any PHP version. Thanks!

Comment (bocharsky-bw) [inline src/AppBundle/Form/PostType.php]: CI with `Travis` will be a great!

**Comment by ben29** [comment]
i can do for Hebrew
still needed?

**Pull request merged**